sqlglot/optimizer/qualify_tables.py

```diff
@@ -74,6 +77,32 @@ def _qualify(table: exp.Table) -> None:
             if isinstance(node, exp.Table) and node.name not in cte_names:
                 _qualify(node)
 
+    canonical_aliases: t.Dict[str, str] = {}
+
+    def _set_alias(
```

Comment:
The purpose of this helper is to consolidate alias-setting logic into one place. We previously used to repeat `exp.alias_` calls in various places that you can find below in the diff. Moreover, we were also slightly inconsistent in the way we did that. Its arguments are needed to mimic the previous control flow.

A helpful side-effect of this is that it makes storing the old -> new alias mapping easy.